config/defaultMappingTemplate.json

```diff
@@ -7,117 +7,103 @@
-              "type" : "string"
+              "type" : "keyword"
             },
             "match" : "*",
             "match_mapping_type" : "float"
           }
         }, {
           "double_fields" : {
             "mapping" : {
-              "index" : "not_analyzed",
               "doc_values" : true,
-              "type" : "string"
+              "type" : "keyword"
             },
             "match" : "*",
             "match_mapping_type" : "double"
           }
         }, {
           "byte_fields" : {
             "mapping" : {
-              "index" : "not_analyzed",
               "doc_values" : true,
-              "type" : "string"
+              "type" : "keyword"
             },
             "match" : "*",
             "match_mapping_type" : "byte"
           }
         }, {
           "short_fields" : {
             "mapping" : {
-              "index" : "not_analyzed",
               "doc_values" : true,
-              "type" : "string"
+              "type" : "keyword"
             },
             "match" : "*",
             "match_mapping_type" : "short"
           }
         }, {
           "integer_fields" : {
             "mapping" : {
-              "index" : "not_analyzed",
               "doc_values" : true,
-              "type" : "string"
+              "type" : "keyword"
             },
             "match" : "*",
             "match_mapping_type" : "integer"
           }
         }, {
           "long_fields" : {
             "mapping" : {
-              "index" : "not_analyzed",
               "doc_values" : true,
-              "type" : "string"
+              "type" : "keyword"
             },
             "match" : "*",
             "match_mapping_type" : "long"
           }
         } ],
         "properties" : {
           "category" : {
-            "type" : "string",
-            "index" : "not_analyzed",
+            "type" : "keyword",
             "doc_values" : true
           },
           "hostname" : {
-            "type" : "string",
-            "index" : "not_analyzed",
+            "type" : "keyword",
             "doc_values" : true
           },
           "processid" : {
-            "type" : "string",
-            "index" : "not_analyzed",
+            "type" : "keyword",
             "doc_values" : true
           },
           "processname": {
-            "type" : "string",
-            "index" : "not_analyzed",
+            "type" : "keyword",
             "doc_values": true
           },
           "severity" : {
-            "type" : "string",
-            "index" : "not_analyzed",
+            "type" : "keyword",
             "doc_values" : true
           },
           "source" : {
-            "type" : "string",
-            "index" : "not_analyzed",
+            "type" : "keyword",
             "doc_values" : true
           },
           "summary" : {
-            "type" : "string"
+            "type" : "text",
+            "fielddata": true
```

Comment:
In order to get our aggregation tests working properly, we had to add the fielddata to true. Thoughts?

Reply:
Where to we aggregate on the summary field? Seems like a memory killer according to: 
https://www.elastic.co/guide/en/elasticsearch/reference/current/fielddata.html

Reply:
I really don't like the idea of enabling fielddata, We'll still be able to search on the summary if it's set to Text based on any token within the field, but I don't know that we do any aggregations now based on single words in summary only searches for strings within the summary.

Reply:
Per https://www.elastic.co/guide/en/elasticsearch/reference/current/text.html they recommend using keyword field for sorting/aggregation  - not text fields. I can't imagine aggregating on summary would be very beneficial if it returned every token in the field in an aggregation (like returning new and york rather than new york)

Reply:
@jeffbryner We're aggregating based on summary in a lot of our generic alerts.

Reply:
ES aggregation or python aggregation? ala: https://github.com/mozilla/MozDef/blob/master/alerts/lib/alerttask.py#L243

Reply:
Ahh, you're right, we don't do any ES aggregation in the alerts, I forgot that the aggregation in the alerts is python based. I think I can fix that test to use another keyname and will remove the fielddata as well from the mapping.

Reply:
Yeah, IIRC we did it that way because the ES aggregation made it difficult to get to the actual records themselves and we only deal with small data sets; so it's easier to grab the raw record set and let python aggregate.